cluster/shard_writer.go

```diff
@@ -61,9 +62,20 @@ func (w *ShardWriter) WriteShard(shardID, ownerID uint64, points []models.Point)
 		conn.Close() // return to pool
 	}(conn)
 
+	// Determine the location of this shard and whether it still exists
+	db, rp, sgi := w.MetaClient.ShardOwner(shardID)
```

Comment:
does this re-introduce the same type of race condition with meta data that we had at the data node previously?  I'm not sure it matters that much, but just curious.